tests/src/ExistingSiteJavascript/CollaboraIntegrationTest.php

```diff
@@ -30,26 +31,97 @@ public function testCollaboraPreview(): void {
         ]);
         $this->drupalLogin($user);
         $media = $this->createDocumentMedia('Shopping list', 'shopping-list', 'Chocolate, pickles');
+
         $this->drupalGet('/cool/view/' . $media->id());
+
         $this->getSession()->switchToIFrame('collabora-online-viewer');
+        $this->assertCollaboraDocumentCanvas();
+        $this->assertCollaboraDocumentName('shopping-list.txt');
+        $this->assertCollaboraWordCountString('2 words, 18 characters');
 
-        $assert_session = $this->assertSession();
-        $canvas = $assert_session->waitForElement('css', 'canvas#document-canvas');
-        $this->assertNotNull($canvas, 'The canvas element was not found after 10 seconds.');
+        // Verify the read-only mode.
+        $readonly_indicator = $this->assertWaitForElement('.status-readonly-mode');
+        $this->assertSame('Read-only', $readonly_indicator->getText());
```

Comment:
We might run into the old stale reference problem with selenium here.
Or it might never occur, if this part of the page is more stable.
I'd say we keep this one simple, and change it if the problem ever occurs.